### PR TITLE
TW-217: update new UX for share and forward screen

### PR DESCRIPTION
**Pull request by sherlockvn**
### Demo mobile

https://github.com/linagora/twake-on-matrix/assets/43041967/672903f8-10f3-48e5-82ba-32bdcbc60ac2

### Demo Web:

 

https://github.com/linagora/twake-on-matrix/assets/43041967/64bb89e0-6139-4c50-bc54-a081487f6afd

### CODE REVIEW

File: lib/pages/forward/recent_chat_list.dart [hunk 1]
```diff
@@ -11,66 +11,77 @@ class RecentChatList extends StatelessWidget {
...
     required this.onSelectedChat,
+    required this.selectedChatNotifier,
   });
 
   final List<Room> rooms;
 
   final ScrollController recentChatScrollController;
 
-  final ValueNotifier<List<String>> selectedEventsNotifier;
+  final ValueNotifier<String> selectedChatNotifier;
```
Comment (nqhhdev) [inline lib/pages/forward/recent_chat_list.dart:22]: Forgot dispose `selectedChatNotifier`

File: lib/pages/forward/recent_chat_list.dart [hunk 2]
```diff
@@ -11,66 +11,77 @@ class RecentChatList extends StatelessWidget {
...
         return ListView.builder(
           padding: EdgeInsets.zero,
           shrinkWrap: true,
           controller: recentChatScrollController,
           itemCount: rooms.length,
           itemBuilder: (BuildContext context, int index) {
             final room = rooms[index];
-            final selected = selectedEvents.contains(room.id);
+            final selected = selectedChat == room.id;
             return Material(
```
Comment (nqhhdev) [inline lib/pages/forward/recent_chat_list.dart:39]: IMO: No need, can you remove it?

File: lib/pages/forward/recent_chat_list.dart [hunk 3]
```diff
@@ -11,66 +11,77 @@ class RecentChatList extends StatelessWidget {
...
-                    ),
-                    Avatar(
-                      mxContent: room.avatar,
-                      name: room.getLocalizedDisplayname(
-                        MatrixLocals(L10n.of(context)!),
+                child: Padding(
+                  padding: const EdgeInsetsDirectional.only(
+                    top: 8.0,
+                    bottom: 8.0,
+                  ),
```
Comment (nqhhdev) [inline lib/pages/forward/recent_chat_list.dart:48]: IMO: It's really better when you move to a style file

File: lib/pages/forward/forward.dart [hunk 4]
```diff
@@ -46,7 +45,7 @@ class ForwardController extends State<Forward> with SearchRecentChat {
   final AutoScrollController recentChatScrollController =
       AutoScrollController();
 
-  final selectedEventsNotifier = ValueNotifier(<String>[]);
+  final ValueNotifier<String> selectedChatNotifier = ValueNotifier('');
```
Comment (hoangdat) [inline lib/pages/forward/forward.dart:48]:
```suggestion
  final ValueNotifier<String> selectedRoomIdNotifier = ValueNotifier('');
```

**Review by hoangdat — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**